### ISSUE TITLE
Nightly link failures with Threads/Pthreads backend

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#4848]
Following merge of PR #4831 builds that enable Pthread/Threads backend fail to complete linking with output:

```
 97%] Linking CXX executable KokkosAlgorithms_UnitTest_StdAlgoCompileOnly
cd /ascldap/users/ndellin/kokkos/testing/Blake-Gcc/algorithms/unit_tests && /home/projects/x86-64/cmake/3.19.3/bin/cmake -E cmake_link_script CMakeFiles/KokkosAlgorithms_UnitTest_StdAlgoCompileOnly.dir/link.txt --verbose=1
/home/projects/x86-64/gcc/7.2.0/bin/g++ -g -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -g -DKOKKOS_DEPENDENCE -march=skylake-avx512 -mtune=skylake-avx512 -mrtm CMakeFiles/KokkosAlgorithms_UnitTest_StdAlgoCompileOnly.dir/TestStdAlgorithmsCompileOnly.cpp.o -o KokkosAlgorithms_UnitTest_StdAlgoCompileOnly  ../../containers/src/libkokkoscontainers.a ../../core/src/libkokkoscore.a /usr/lib64/libdl.so
../../core/src/libkokkoscore.a(Kokkos_ThreadsExec_base.cpp.o): In function `std::thread::thread<void (&)()>(void (&)())':
/home/projects/x86-64/gcc/7.2.0/include/c++/7.2.0/thread:122: undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
make[2]: *** [algorithms/unit_tests/KokkosAlgorithms_UnitTest_StdAlgoCompileOnly] Error 1
make[2]: Leaving directory `/home/ndellin/kokkos/testing/Blake-Gcc'
make[1]: *** [algorithms/unit_tests/CMakeFiles/KokkosAlgorithms_UnitTest_StdAlgoCompileOnly.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

I tested with changes from PR #4841 but those changes did not resolve this issue for me

Sample reproducer:
```
module load cmake/3.19.3 gcc/7.2.0

# cmake
cmake -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_FLAGS="-g -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized" -DKokkos_ENABLE_PTHREAD=ON -DKokkos_ARCH_SKX=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DKokkos_ENABLE_TESTS=ON -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_CXX_STANDARD=14 -DCMAKE_BUILD_TYPE=DEBUG -DKokkos_ENABLE_DEBUG=ON -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF $KOKKOS_PATH
```

**Comment by masterleinad** [comment]
Hmm... I can reproduce the failure on `develop` but for me it's fixed by #4841.
Would you mind checking which branch in [cmake/Modules/FindTPLTHREADS.cmake](https://github.com/kokkos/kokkos/pull/4841/files#diff-c2b62cffd37625c15dd87c15509a743cbfc3cfde68800e7eae296d8643fe2c73) you are using?

Does replacing that file with the content of [cmake/Modules/FindTPLPTHREAD.cmake](https://github.com/kokkos/kokkos/pull/4841/files#diff-bd698488b2e9c5447afcaf177087ccee9f241b2822299d3647e4b3b36b5898a0) work instead?

**Comment by ndellingwood** [comment]
I retested with #4841 and gcc and intel builds worked for me, I must have had a dirty directory
